modules/porous_flow/src/materials/PorFlowMaterialMassFractionBuilder.C

```diff
@@ -0,0 +1,120 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+
+#include "PorFlowMaterialMassFractionBuilder.h"
+
+#include "Conversion.h"
+
+template<>
+InputParameters validParams<PorFlowMaterialMassFractionBuilder>()
+{
+  InputParameters params = validParams<Material>();
+
+  params.addRequiredRangeCheckedParam<unsigned int>("num_phases", "num_phases>0", "The number of fluid phases in the simulation");
+  params.addRequiredRangeCheckedParam<unsigned int>("num_components", "num_components>0", "The number of fluid components in the simulation");
+  params.addCoupledVar("mass_fraction_vars", "List of variables that represent the mass fractions.  Format is 'f_ph0^c0 f_ph0^c1 f_ph0^c2 ... f_ph0^c(N-1) f_ph1^c0 f_ph1^c1 fph1^c2 ... fph1^c(N-1) ... fphP^c0 f_phP^c1 fphP^c2 ... fphP^c(N-1)' where N=num_components and P=num_phases, and it is assumed that f_ph^cN=1-sum(f_ph^c,{c,0,N-1}) so that f_ph^cN need not be given.  If no variables are provided then num_phases=1=num_components.");
+  params.addRequiredParam<UserObjectName>("PorFlowVarNames_UO", "The UserObject that holds the list of Porous-Flow variable names.");
+  params.addClassDescription("This Material forms a std::vector<std::vector ...> of mass-fractions out of the individual mass fractions");
+  return params;
+}
+
+PorFlowMaterialMassFractionBuilder::PorFlowMaterialMassFractionBuilder(const InputParameters & parameters) :
+    DerivativeMaterialInterface<Material>(parameters),
+
+    _num_phases(getParam<unsigned int>("num_phases")),
+    _num_components(getParam<unsigned int>("num_components")),
+    _porflow_name_UO(getUserObject<PorFlowVarNames>("PorFlowVarNames_UO")),
+
+    _mass_frac(declareProperty<std::vector<std::vector<Real> > >("PorFlow_mass_frac")),
+    _mass_frac_old(declarePropertyOld<std::vector<std::vector<Real> > >("PorFlow_mass_frac")),
+    _dmass_frac_dvar(declareProperty<std::vector<std::vector<std::vector<Real> > > >("dPorFlow_mass_frac_dvar")),
+
+    _num_passed_mf_vars(coupledComponents("mass_fraction_vars"))
+{
+  if (_num_passed_mf_vars != _num_phases*(_num_components - 1))
+    mooseError("PorFlowMaterialMassFractionBuilder: The number of mass_fraction_vars is " << _num_passed_mf_vars << " which must be equal to num_phases (" << _num_phases << ") multiplied by num_components-1 (" << _num_components - 1 << ")");
+
+  _mf_vars.resize(_num_passed_mf_vars);
```

Comment:
move all these resizes into the initializer list
`    _mf_vars(_num_passed_mf_vars),`